### PR TITLE
feat: add telemetry

### PR DESCRIPTION
**Pull request by SgtPooki**
fixes https://github.com/ipfs-shipyard/ipfs-check/issues/28

### CODE REVIEW

**Comment by SgtPooki** [comment]
ping @djmcquillan @whizzzkid

**Review by djmcquillan — APPROVED** [review]

**Comment by SgtPooki** [comment]
@aschmahmann any comments before i merge this? How do we deploy this once merged?

**Review by aschmahmann — COMMENTED** [review]
@SgtPooki not much to add here. Given the nature of the PR someone on your team is probably much better equipped to review than me.

The website itself should be auto-deployed by Fleek when you merge to master (it's managed by a PL account). I think it currently just deploys the "web" folder and doesn't need to do anything special, if that's changed as a function of this PR then you might need to change the Fleek configuration as well.

File: web/index.html
Comment (aschmahmann) [inline web/index.html:9]: Forgive my ignorance here, but why do these have `/` in front now and require running npm?

Any reason things can't be left simply where you can just open up `web/index.html` and everything just works?

File: README.md
Comment (aschmahmann) [inline README.md:46]: Related to my question below, is this necessary? Can't you just open up index.html in your browser (with the backendUrl set to be local, or deployed elsewhere) and you're good to go.

File: web/index.html
Comment (SgtPooki) [reply web/index.html:9]: The / is not related to the build, just, more idiomatic for representing relative asset.

The reason things can't be "just open up web/index.html and everything works" is because our ignite-metrics package isn't fully exporting browser compatible esm imports at the moment, so we require a build step.

We could commit a minified telemetry.min.js file and include that, but i'd rather we didn't.

File: README.md
Comment (SgtPooki) [reply README.md:46]: Unfortunately, the telemetry.js file has to be built in order for things to load properly. We could solve this with properly exported esm similar to how @whizzzkid did in ipfs-geoip, but the work has not been done for that yet.

I could fix things to allow users to just load web/index.html and have things work without a build step but will need to add a no-op web/dist/telemetry.js. I can do that.

Our deployment of ipfs-check will still require the build step though

File: web/index.html
Comment (aschmahmann) [reply web/index.html:9]: > The / is not related to the build, just, more idiomatic for representing relative asset.

You might know better than me here, but this idiom doesn't really work if you just load index.html locally. Instead everything breaks as it tries to load from `/` on your machine using `file://` (at least on Chrome + Windows).

When I remove the `/` and load everything seems to work fine, and I don't have any console errors. Perhaps the telemetry isn't reporting properly then but idk.

File: README.md
Comment (aschmahmann) [reply README.md:46]: Ah ok, you need to build a minified telemetry.js with all the included dependencies and such. Makes sense, doesn't seem reasonable to block on you doing more work here. At some point this whole tool probably merges with pl-diagnose anyway #24 and has lots of fancy JS build tooling anyhow.

File: web/telemetry.js
Comment (SgtPooki) [reply web/telemetry.js:5]: since these are only temporary (removed when https://github.com/ipfs-shipyard/ipfs-check/issues/29 is done) I don't think so, but I'm open

File: README.md
Comment (SgtPooki) [reply README.md]: Updated `web/dist/telemetry.js` to commit with an empty script file (comment only) and addressed some updates in the readme to be more accurate.

**Comment by djmcquillan** [comment]
lgtm with added telemetry.min.js

**Pull request merged**
